src/content/app/genome-browser/state/track-config/trackConfigSlice.ts

```diff
@@ -68,54 +128,247 @@ export const updateTrackStatesAndSave: ActionCreator<
   browserStorageService.saveTrackStates(trackStates);
 };
 
+export const prepareTrackConfigs = (trackCategories: GenomeTrackCategory[]) => {
+  const defaultTrackConfigs: TrackConfigs = {};
+
+  defaultTrackConfigs['gene-focus'] = {
+    showSeveralTranscripts: false,
+    showTranscriptIds: false,
+    showTrackName: false,
+    showFeatureLabel: true,
+    trackType: TrackType.GENE
+  };
+
+  trackCategories.forEach((category) => {
+    category.track_list.forEach((track) => {
+      const trackId = track.track_id.replace('track:', '');
+      const trackType = getTrackType(trackId);
+
+      if (trackType === TrackType.GENE) {
+        defaultTrackConfigs[trackId] = getDefaultGeneTrackConfig();
+      } else {
+        defaultTrackConfigs[trackId] = getDefaultRegularTrackConfig();
+      }
+    });
+  });
+  return { tracks: defaultTrackConfigs };
+};
+
+export const getTrackConfigsForGenome = (
+  genomeId: string,
+  trackCategories: GenomeTrackCategory[]
+): TrackConfigsForGenome => {
+  return genomeId
+    ? {
+        ...defaultTrackConfigsForGenome,
+        ...prepareTrackConfigs(trackCategories),
+        ...getPersistentTrackConfigsForGenome(genomeId)
+      }
+    : defaultTrackConfigsForGenome;
+};
+
+export const getPersistentTrackConfigsForGenome = (
+  genomeId: string
+): Partial<TrackConfigsForGenome> => {
+  const trackConfigs = browserTrackConfigStorageService.getTrackConfigs();
+
+  if (!genomeId || !trackConfigs[genomeId]) {
+    return {};
+  }
```

Comment:
this check probably can go: `genomeId` is defined as string.

`return trackConfigs[genomeId] ?? {};`